src/transpiling/jsx_precompile.rs

```diff
@@ -0,0 +1,1830 @@
+                    " {}=\"{}\"",
+                    escape_html(&attr_name).as_str(),
+                    escape_html(string_lit.value.as_ref()).as_str()
+                  );
+
+                  strings.last_mut().unwrap().push_str("");
+                  strings
+                    .last_mut()
+                    .unwrap()
+                    .push_str(serialized_attr.as_str());
+                }
+                // I've never seen this being possible as it would
+                // always be treated as an expression.
+                Lit::Bool(_) => {}
+                Lit::Null(_) => {}
+                Lit::Num(_) => {}
+                Lit::BigInt(_) => {}
+                Lit::Regex(_) => {}
+                Lit::JSXText(_) => {}
+              },
+              JSXAttrValue::JSXExprContainer(jsx_expr_container) => {
+                strings.last_mut().unwrap().push(' ');
+                strings.push("".to_string());
+                match &jsx_expr_container.expr {
+                  // This is treated as a syntax error in attributes
+                  JSXExpr::JSXEmptyExpr(_) => {}
+                  JSXExpr::Expr(expr) => {
+                    let call_expr = self
+                      .convert_to_jsx_attr_call(&attr_name.to_string(), expr);
+                    dynamic_exprs.push(Expr::Call(call_expr));
+                  }
+                }
+              }
+              // These makes no sense on as attribute on HTML elements
+              // so we ignore them.
+              JSXAttrValue::JSXElement(_) => {}
+              JSXAttrValue::JSXFragment(_) => {}
+            }
+          }
+          // This case is already handled earlier
+          // Case: <div {...props} />
+          JSXAttrOrSpread::SpreadElement(_) => {}
+        };
+      }
+    }
+
+    strings.last_mut().unwrap().push('>');
+
+    // There are no self closing elements in HTML, only void elements.
+    // Void elements are a fixed list of elements that cannot have
+    // child nodes.
+    // See https://developer.mozilla.org/en-US/docs/Glossary/Void_element
+    // Case: <br /> -> <br>
+    // Case: <meta /> -> <meta>
+    if is_void_element(&name) {
+      // Since self closing tags don't exist in HTML we don't need to
+      // add the "/" character. If the "/" character is present,
+      // browsers will ignore it anyway.
+      return;
+    }
+
+    self.serialize_jsx_children_to_string(&el.children, strings, dynamic_exprs);
+
+    let closing_tag = format!("</{}>", name);
+    strings.last_mut().unwrap().push_str(closing_tag.as_str());
+  }
+
+  fn gen_template(
+    &mut self,
+    template_index: usize,
+    static_strs: Vec<String>,
+    dynamic_exprs: Vec<Expr>,
+  ) -> Expr {
+    let name = create_tpl_binding_name(template_index);
+    self.templates.push((template_index, static_strs));
+
+    let mut args: Vec<ExprOrSpread> =
+      Vec::with_capacity(1 + std::cmp::max(1, dynamic_exprs.len()));
```

Comment:
Why do an std::cmp::max here? Looks like it should just be `1 + dynamic_exprs.len()`?